api/controllers/view-homepage.js

```diff
@@ -36,23 +36,20 @@ module.exports = {
 
     // Otherwise IWMIH, we know the current user is logged in.
     // So look up the record in the database that represents this logged in user.
-    User.findOne({
+    var loggedInUserRecord = await User.findOne({
       id: req.session.userId
-    }).exec(function (err, loggedInUserRecord) {
-      if (err) { return exits.error(err); }
+    });
 
-      if (!loggedInUserRecord) {
-        return exits.error(new Error('Consistency violation: The requesting user is logged in as user `'+req.session.userId+'`, but no such user exists in the database.  This should never happen!'));
-      }
-
-      // Then respond with HTML.
-      return exits.success({
-        username: loggedInUserRecord.username,
-        remark: loggedInUserRecord.remark,
-        avatarColor: loggedInUserRecord.avatarColor
-      });
+    if (!loggedInUserRecord) {
+      return exits.error(new Error('Consistency violation: The requesting user is logged in as user `'+req.session.userId+'`, but no such user exists in the database.  This should never happen!'));
```

Comment:
this can just be `throw …` now, instead of `return exits.error(…`

(because now that we don't have asynchronous callbacks everywhere, we don't have to worry about throwing an error crashing the process!)